design/sketch.webidl

```diff
@@ -368,13 +368,11 @@ interface WebGPUInputStepMode {
 
 dictionary WebGPUVertexAttributeDescriptor {
     u32 shaderLocation;
-    u32 inputSlot;
+    u32 vertexBufferIndex;
     u32 offset;
-    WebGPUVertexFormatEnum format;
 };
 
-dictionary WebGPUVertexInputDescriptor {
-    u32 inputSlot;
```

Comment:
I'd be in favor of keeping this, so that applications can define "holes" in their vertex buffers, which would allow minimizing state setting between two pipeline using very similar but different vertex buffers (they could match on VB 0 to 5 and one of them having VB 6 and the other VB 7).

Actually having holes in interfaces like this is something I also wanted to bring up for the pipeline layout and color attachments as well. We already had at least on project prototyping on Dawn ask if we could have holes in the pipeline layout.

Reply:
JavaScript arrays are sparse and heterogeneous, so they don't prevent us from having holes. With the change at [1], we could probably (if the IDL accepts it) have:

```js
{ vertexBuffers: [buffer0, null, buffer2] }
```

or

```
const vertexBuffers = [];
vertexBuffers[0] = buffer0;
vertexBuffers[2] = buffer2;
{ vertexBuffers }
```